tfhe/build.rs

```diff
@@ -73,7 +73,71 @@ fn gen_c_api() {
         .write_to_file(output_file);
 }
 
+fn panic_if_non_optimized_build() {
+    // This has a lot of \n and \ to control formatting precisely
+    const ERROR_MESSAGE: &str = "\
+    It seems that tfhe-rs is being compiled with insufficient optimization level.\n\
+    FHE is already slow on its own, and so compiler optimizations have to be turned on.\n\
+    \n\
+    By default rust/cargo does not enable optimizations, make sure you build/run with the \n\
+    release flag:\n    \
+    - `cargo run --release`\n    \
+    - `cargo build --release`\n\
+    (Note that doing `cargo build --release && cargo run`, will build in release mode\n\
+     but then rebuild in debug mode and run in debug mode)\n
+    \n\
+    * If you wish to be able to use a debugger, consider adding the following lines \n\
+    to the projects's Cargo.toml and compile in release\n\
+    \t```\n\
+    \t[profile.release]\n\
+    \tdebug = true\n\
+    \t```\n\
+    \t\n\
+    * It is also possible to override the opt-level only for tfhe-rs in the Cargo.toml\n\
+    \t```\n\
+    \t[profile.dev.package.tfhe]\n\
+    \topt-level=3\n\
+    \t```\n\
+    \tMore generally:\n\
+    \t```\n\
+    \t[profile.profile_name.package.tfhe]\n\
+    \topt-level=3\n\
+    \t```\n\
+    \t\n\
+    * If you really need a non optimized / less optimized build set the environment variable:\n\
+    \t```\n\
+    \texport TFHE_RS_ALLOW_NON_OPTIMIZED_BUILD=true\n\
+    \t```\n\
+    Or in .cargo/config.toml add\n\
+    \t```\n\
+    \t[env]\n\
+    \tTFHE_RS_ALLOW_NON_OPTIMIZED_BUILD=\"true\"\n\
+    ```\n\
+";
+    if option_env!("CARGO_PRIMARY_PACKAGE").is_none() {
+        // If CARGO_PRIMARY_PACKAGE is set, then we are build tfhe-rs is not being built
+        // as a dependency.
```

Comment:
```
       // If CARGO_PRIMARY_PACKAGE is set, then tfhe-rs is not being built
        // as a dependency.
```

Reply:
maybe I should use that for the C API thing from last time

Reply:
we need to set the escape hatch in our Makefile though right ?

Reply:
CARGO_PRIMARY_PACKAGE  is used so that we don't have to use optimized build / set the env var to allow them as we know what we are doing 😬, so we are exempt of the opt-level check

Reply:
ha nice

Reply:
could you maybe update the code for the force skip cbindgen to use the primary package thing ? in a separate commit but as it's the same kind of idea it could be good to do at the same time ?

Reply:
though I don't know if that would work for Google who use rustc directly 🤔

Reply:
Also I hope Cargo will not get confused with the two tfhe crates... maybe better in a separate PR then

Reply:
I'll experiment with the primary package thing to remove the cbindgen related feature